File: es_11.cpp
// 11. Chiedere in input un numero n positivo (con controllo),
// stampare a video tutti i numeri da 0 a n.

#include <iostream>
using namespace std;

int main() {
    int n;

    do {
        cout << "Inserisci un numero positivo: ";
        cin >> n;
    } while (n < 0);

    for (int i = 0; i <= n; i++) {
        cout << i << endl;
    }

    return 0;
}

File: Python/es_11.py
from itertools import count

# 11. Chiedere in input un numero n positivo (con controllo),
# stampare a video tutti i numeri da 0 a n.

while True:
    n = int(
        input('Inserisci un numero n > 0: ')
    )
    
    if n > 0: break
    else: print('Numero non valido')
    
for i in count (start=0, step=1):
    print(
        f'{i}'      # "0, 1, 2, 3, 4 ...\n"
    )
    
    if i >= n: break